### ISSUE TITLE
[BUG/ISSUE] Dry deposition flux diagnostic does not work for full-PBL mixing scheme

### ISSUE THREAD

**Issue by ltmurray** [geoschem/geos-chem#488]
The full-PBL dry deposition fluxes are calculated in the DO_TEND subroutine in GeosCore/mixing_mod.F90, so the call to
```
    IF ( State_Diag%Archive_DryDepMix .or.  &
            State_Diag%Archive_DryDep        ) THEN
          State_Diag%DryDepMix = 0.0_f4
    ENDIF
```
that occurs after the call to DO_TEND erases the fluxes. Moving that call to before either the non-local PBL or full-PBL mixing schemes re-enables the dry deposition diagnostic for the full-PBL mixing scheme, as in https://github.com/ltmurray/geos-chem/tree/turbday_nc_bugfix

However, the fluxes calculated are ~40-50% lower than the fluxes from the non-local scheme, as well as comparably lower than the E-D tendencies in the budget diagnostic using the full-PBL scheme for species that have no emissions (e.g., 210Pb). So there may still be an issue with the frequency of the diagnostic, since we used to archive dry deposition over the chemistry time step, rather than the dynamic time step?

**Comment by lizziel** [comment]
Thanks @ltmurray. It looks like this bug has been in there since 12.1.0 and introduced during refactoring of unrelated code (my bad). Prior to that the values were set to zero at the start of DO_MIXING and before calls to either of the PBL mixing subroutines and DO_TEND. This fix will go into 13.0.0.

Regarding additional problems with the diagnostic related to the update frequency, we will look into it.

**Comment by lizziel** [comment]
Looking closer at this, I don't think this fix will work with the version in development. `State_Diag%DryDepMix` is now also set in the call to `Compute_Sflx_for_Vdiff` which is right before the call to `DO_MIXING` in `main.F`. Zeroing out the array anywhere in `DO_MIXING` therefore erases what was set in `Compute_Sflx_for_Vdiff`.

The best solution may be to tap into currently unused subroutine `Zero_Diagnostics_StartofTimestep` in `diagnostics_mod.F90`. That subroutine is a placeholder for if we ever needed to cleanly zero out a diagnostic array at the start of each timestep.

There is a plan to separate out dry deposition, mixing, and emissions since the current way it is written, such that they are intertwined, is very confusing. It also makes the budget diagnostics hard to interpret. @yantosca will be working on that update for GEOS-Chem 13.1. See feature request https://github.com/geoschem/geos-chem/issues/327.

**Comment by lizziel** [comment]
This is now fixed in https://github.com/geoschem/geos-chem/commit/58ea3708509ef8515837bda4f341847853106999. It only addresses the issue of where the array is set to zero. More investigation will be needed to see if there are any further issues.

We previously addressed the issue of the netcdf update timestep vs binary timestep mismatch by splitting up the dry dep diag into DryDepMix and DryDepChm, and summing them to get State_Diag%DryDep. DryDepMix is updated every convection timestep which is the same as the frequency for diagnostic accumulation. DryDepChm is only updated every chemistry timestep, but starting with the very first timestep of the run. DryDepChm should NOT get reset to zero so the same value would be used for two diagnostic timesteps in a row, and in theory allow replication of the binary diagnostic values. However, this was done quite a while ago so perhaps something got changed along the way introducing a bug, as was done for the DryDepMix zero assigment.

**Comment by msulprizio** [comment]
I am closing out the issue since this fix is going into 13.0.0. If problems remain, a separate Github issue should be opened.
